src/components/OrgLogoUpload/OrgLogoUpload.tsx

```diff
@@ -0,0 +1,167 @@
+}) => {
+  const fileInput = React.createRef<HTMLInputElement>();
+
+  const { imageUrl, formFileUploadProps } = useImageUploader(
+    getAvatarPathWithFallback(original, name)
+  );
+
+  const { showError } = useApeSnackbar();
+
+  const onInputChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files.length) {
+      if (!VALID_FILE_TYPES.includes(e.target.files[0].type)) {
+        showError(
+          e.target.value +
+            ' is invalid, allowed files are: ' +
+            VALID_FILE_TYPES.join(', ')
+        );
+        formFileUploadProps.onChange(undefined);
+      } else {
+        formFileUploadProps.onChange(e.target.files[0]);
+      }
+    }
+  };
+  const uploadLogo = async (orgId: number, logo: File) => {
+    const image_data_base64 = await fileToBase64(logo);
+    await updateOrgLogo(orgId, image_data_base64);
+  };
+
+  const onSave = async () => {
+    commit(true);
+    if (formFileUploadProps.value === undefined || !commit) {
+      return;
+    }
+    await uploadLogo(id, formFileUploadProps.value);
+    commit(false);
+  };
+
+  useEffect(() => {
+    const cropImage = async (imageUrl: string) => {
+      if (formFileUploadProps.value) {
+        const canvas = await centerCropImage(imageUrl);
+        const data = canvas.toDataURL();
+        const blobBin = atob(data.split(',')[1]);
+        const mimeString = data.split(',')[0].split(':')[1].split(';')[0];
+        const array = [];
+        for (let i = 0; i < blobBin.length; i++) {
+          array.push(blobBin.charCodeAt(i));
+        }
+        const blob = new Blob([new Uint8Array(array)], { type: mimeString });
+        const file = new File([blob], formFileUploadProps.value.name);
+        formFileUploadProps.onChange(file);
+      }
+    };
+    if (formFileUploadProps.value) {
+      cropImage(imageUrl).catch(console.warn);
+    }
+  }, [formFileUploadProps.hasChanged]);
+
+  return (
+    <Box css={{ position: 'relative' }}>
+      <Avatar
+        path={imageUrl}
+        orgLogo={true}
+        name={name}
+      />
+      <Box css={{ position: 'absolute', bottom: '0', right: '0' }}>
+        <Box css={{ display: 'flex', flexDirection: 'row' }}>
+          {formFileUploadProps.hasChanged && !!commit && (
+            <Button onClick={onSave} size="smaller">
+              {<UploadIcon />}
+            </Button>
+          )}
+          {formFileUploadProps.hasChanged && (
+            <Button
+              onClick={() => {
+                formFileUploadProps.onChange(undefined);
+                // This clears the fileInput so that if the user chooses the same file again, onChange still fires
+                if (fileInput.current) {
+                  fileInput.current.value = '';
+                }
+              }}
+              size="smaller"
+            >
+              {<CloseIcon />}
+            </Button>
+          )}
+          {isAdmin && !formFileUploadProps.hasChanged &&(
```

Comment:
formatting seems to be off here. there should be a git pre-commit hook that fixes this; if you have that disabled for some reason, you should run `yarn lint:fix` and `yarn prettier:fix` before committing